QRCoder/QRCodeData.cs

```diff
@@ -21,11 +21,12 @@ public QRCodeData(int version)
             for (var i = 0; i < size; i++)
                 this.ModuleMatrix.Add(new BitArray(size));
         }
-#if NETFRAMEWORK || NETSTANDARD2_0 || NET5_0
+
+        [Obsolete("This constructor will be removed in version 2.0.0.")]
```

Comment:
Marked as obsolete, because file handling should be done by the calling application? Or for what reason? (Don't get me wrong - no critique - I just want to understand the reason. :-) )

Reply:
My goal was to eliminate ifdefs.  Here the method did not exist on the most recent platform (.NET 6).  If this was accidental, then we can just remove the ifdef and leave it available for all platforms.  On the other hand, if the intent was to remove this method from future versions, I suggest we just mark it obsolete and remove in v2 (with no ifdefs).

Personally, I am not much in favor of methods to read/write directly from the file system, favoring instead for methods that read/write to `Stream` instances.  I'm probably biased, as I'm typically writing server applications that is going to read/write data in an Azure blob or send over HTTP.   But whatever you think is fine with me.

So, just lmk if I should remove the `[Obsolete]` attribute or not.

Reply:
>  If this was accidental, then we can just remove the ifdef and leave it available for all platforms.

Yes, I would say that was accidental. 7 years ago, when I was implementing this, QRCoder targeted (besides others) also PCL (is this still a thing?). The PCL library back then had no File support if I remember correctly, thus I added `#if !PCL` as flag. See here: https://github.com/codebude/QRCoder/blame/a45f953488aa7bc9b95308ffeafa0cc4abefcc82/QRCoder/QRCodeData.cs#L24

5 years ago switching from PCL to .NET Standard the term was changed to `#if !NETSTANDARD1_1` (because File.WriteAllBytes is only available from .NET Standard 1.3 on):
https://github.com/codebude/QRCoder/blame/3a528d8234ed50e490d8f2176d309607679c88f7/QRCoder/QRCodeData.cs#L24

Later the condition was inverted from exclusion to inclusion `#if NETFRAMEWORK || NETSTANDARD2_0` and then changed multiple times `#if NETFRAMEWORK || NETSTANDARD2_0 || NETSTANDARD2_1` --> `#if NETFRAMEWORK || NETSTANDARD2_0 || NET5_0`.

So the inital intention was to suppress the constructor for environments where File.WriteAllBytes wasn't available. Since we do not target .NET Standard 1.1 any longer (https://github.com/codebude/QRCoder/blob/069e4a14e48a94f0d46a8449e31dcac5205665dc/QRCoder/QRCoder.csproj#L4) I would say just remove the ifdefs *and* the obsolete attribute.

> Personally, I am not much in favor of methods to read/write directly from the file system, favoring instead for methods that read/write to Stream instances.

Back then I just thought if someone doesn't want to deal with files but read from/to memory, the constructor taking byte[] and the method returning byte[] would be convenient enough.

Reply:
```suggestion
```